engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -382,22 +407,27 @@ private void updateJobState(final State newState) {
     statesJobColumnFamily.update(fkJob, jobState);
   }
 
-  private void makeJobActivatable(final DirectBuffer type, final long key) {
+  private void makeJobActivatable(final DirectBuffer type, final long key, final String tenantId) {
     EnsureUtil.ensureNotNullOrEmpty("type", type);
 
     jobTypeKey.wrapBuffer(type);
-
     jobKey.wrapLong(key);
+    if (!TenantOwned.DEFAULT_TENANT_IDENTIFIER.equals(tenantId)) {
+      tenantIdKey.wrapString(tenantId);
+    }
```

Comment:
❌ This will result in bugs. By not wrapping the `tenantIdKey` if it's default, it means the `tenantIdKey` is still wrapped with the value it had previously. So if I first make a job activatable for tenant `X` and try to make another activatable for `<default>`, the second job will now also be activatable for tenant `X`.

Reply:
I noticed this is already changed in a later commit.